### PR TITLE
Refactor valueset handling

### PR DESCRIPTION
**Pull request by lrasmus**
This makes changes to how the program handles value sets and converts them to concept sets.  The main changes are:

1. Primarily use `ConceptCodeCsvFileValuesetService` for reading and loading value sets.
2. With the change in 1, since we are using a CSV, the `VS_TAB` config option does not really apply (this was more of an artifact of using `SpreadsheetValuesetService`).  This config option is now considered optional.  If it is present, that is an indication to the program that it should instead use `SpreadsheetValuesetService`.
3. When loading a value set CSV, the code should be the vocabulary code, not the OMOP CONCEPT_ID.  This was a previous assumption built into the program that has now been changed.  This will allow us to create real value set files from VSAC and other sources.
4. Cache OMOP CONCEPT_ID as we build the phenotype - once the value set codes have all been mapped to corresponding OMOP CONCEPT_IDs, we will write a cache file (named the same as the input CSV file, with a `.cache` extension) that contains the CONCEPT_ID.  In subsequent runs of the same phenotype, we can reuse any known CONCEPT_IDs instead of having to do a full vocabulary search.  This will save on execution time after the first run.

### CODE REVIEW

**Comment by lrasmus** [comment]
Wanted to get main code staged into PR - but note that additional testing will be added (especially since this reduces overall coverage).

**Comment by psbrandt** [comment]
To figure out where coverage is missing you can expand the tree at the bottom of this page: https://coveralls.io/builds/27909085

File: src/main/java/edu/phema/elm_to_omop/io/SpreadsheetWriter.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.commons.csv.CSVPrinter;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Creates a spreadsheet file on disk
+ */
+public class SpreadsheetWriter {
```
Comment (psbrandt) [inline src/main/java/edu/phema/elm_to_omop/io/SpreadsheetWriter.java]: Are we using the term "spreadsheet" and "CSV file" interchangeably?

File: src/main/java/edu/phema/elm_to_omop/api/CohortService.java [hunk 2]
```diff
@@ -45,6 +46,12 @@ public CohortService(Config config) throws CohortServiceException {
 
         this.config = config;
         this.valuesetService = new SpreadsheetValuesetService(omopService, config.getVsFileName(), config.getTab());
+        if (config.isTabSpecified()) {
```
Comment (psbrandt) [inline src/main/java/edu/phema/elm_to_omop/api/CohortService.java]: Not for this PR, but we should probably add some tests at some point for using a spreadsheet with tabs.

File: src/main/java/edu/phema/elm_to_omop/helper/Config.java [hunk 3]
```diff
@@ -221,7 +221,7 @@ private void runPropertyCheck() {
             LOGGER.severe("ERROR - missing parameter SOURCE");
         }
         if (tab == null) {
-            LOGGER.severe("ERROR - missing parameter VS_TAB");
+            LOGGER.severe("INFO - missing optional parameter VS_TAB");
```
Comment (psbrandt) [inline src/main/java/edu/phema/elm_to_omop/helper/Config.java]: Should this be `LOGGER.info()`?

File: src/main/java/edu/phema/elm_to_omop/io/SpreadsheetWriter.java (hunk 1 above)
Comment (psbrandt) [reply src/main/java/edu/phema/elm_to_omop/io/SpreadsheetWriter.java]: Or maybe it's just to be consistent with `SpreadsheetReader`, which actually reads a CSV file? 😔

File: src/test/java/edu/phema/elm_to_omop/phenotype/FilePhenotypeTest.java [hunk 4]
```diff
@@ -30,7 +35,17 @@
     public void setup() throws Exception {
         MockitoAnnotations.initMocks(this);
 
-        lenient().when(omopRepository.getConceptMetadata("1.2.3.4")).thenReturn(new Concept());
+        // mock for method vocabularySearch
+        when(omopRepository.vocabularySearch(anyString(), anyString())).thenAnswer(new Answer<ArrayList<Concept>>() {
```
Comment (psbrandt) [inline src/test/java/edu/phema/elm_to_omop/phenotype/FilePhenotypeTest.java]: I think this approach is better than the WireMock approach I've been using. We should migrate to using this (not in this PR though).

**Comment by psbrandt** [comment]
If you don't merge today, then I might do it for you, so that I can merge mine second and you don't have to deal with potential conflicts :).

**Comment by lrasmus** [comment]
Thank you @psbrandt, for review and merge.  Sorry for not getting back to this yesterday, glad you proceeded w/ merge!

**Pull request merged**